src/06-positron-rag/lib/insert/prepareArticlesCollection.ts

```diff
@@ -0,0 +1,141 @@
+import weaviate from "weaviate-ts-client"
+import { CLASS_NAME } from "../../02-insert-articles"
+
+/**
+ * Prepare a new Weaviate collection for articles and configure its properties and vectorization
+ */
+export async function prepareArticlesCollection() {
+  const client = weaviate.client({
+    scheme: "https",
+    host: "https://weaviate.stg.artsy.systems",
+  })
+
+  const alreadyExists = await client.schema.exists(CLASS_NAME)
+
+  if (alreadyExists) {
+    console.log(`${CLASS_NAME} class already exists, deleting it`)
+    await client.schema.classDeleter().withClassName(CLASS_NAME).do()
+  }
+
+  const classWithProps = {
+    class: CLASS_NAME,
+    vectorizer: "text2vec-openai",
+    moduleConfig: {
+      "text2vec-openai": {
+        model: "text-embedding-3-small",
+        dimensions: 1536,
+        type: "text",
+        vectorizeClassName: false,
+      },
+    },
+    properties: [
+      {
+        name: "internalID",
+        dataType: ["text"],
+        moduleConfig: {
+          "text2vec-openai": {
+            skip: true,
+            // vectorizePropertyName: false,
+          },
+        },
+      },
+      {
+        name: "title",
+        dataType: ["text"],
+        moduleConfig: {
+          "text2vec-openai": {
+            skip: true,
+            // vectorizePropertyName: false,
+          },
+        },
+      },
+      {
+        name: "href",
+        dataType: ["text"],
+        moduleConfig: {
+          "text2vec-openai": {
+            skip: true,
+            // vectorizePropertyName: false,
+          },
+        },
+      },
+      {
+        name: "publishedAt",
+        dataType: ["date"],
+        moduleConfig: {
+          "text2vec-openai": {
+            skip: true,
+            // vectorizePropertyName: false,
+          },
+        },
+      },
+      {
+        name: "byline",
+        dataType: ["text"],
+        moduleConfig: {
+          "text2vec-openai": {
+            skip: true,
+            // vectorizePropertyName: false,
+          },
+        },
+      },
+      // {
+      //   name: "keyword",
+      //   dataType: ["text"],
```

Comment:
I meant to confirm that array-valued fields are handled automatically. Will follow up on this...

Reply:
Believe the schema was unable to recognize an array of string when I tested with version `1.18.0`, requiring me to [join them](https://github.com/artsy/quantum/pull/13/files#diff-172a2080617f7cfe94a242adee40b6fc2711bdc3ffe1f98c86d783771468fc0cR248) into a single string, but haven't tested it since we bumped to `1.24.14`.